src/stream/src/task/barrier_manager/managed_state.rs

```diff
@@ -71,47 +73,58 @@ impl ManagedBarrierState {
         };
 
         if to_notify {
-            let inner = self.epoch_barrier_state_map.remove(&curr_epoch).unwrap();
-
-            let create_mview_progress = self
-                .create_mview_progress
-                .remove(&curr_epoch)
-                .unwrap_or_default()
-                .into_iter()
-                .map(|(actor, state)| CreateMviewProgress {
-                    chain_actor_id: actor,
-                    done: matches!(state, ChainState::Done),
-                    consumed_epoch: match state {
-                        ChainState::ConsumingUpstream(consumed_epoch) => consumed_epoch,
-                        ChainState::Done => curr_epoch,
-                    },
+            let result = self
+                .epoch_barrier_state_map
+                .iter()
+                .find_or_last(|(_, inner)| match inner {
+                    ManagedBarrierStateInner::Issued {
+                        remaining_actors, ..
+                    } => !remaining_actors.is_empty(),
+                    _ => true,
                 })
-                .collect();
+                .map(|(key, _)| *key);
+            let result = match result {
+                Some(epoch) => self.epoch_barrier_state_map.split_off(&epoch),
```

Comment:
`split_off` returns everything after epoch but I think we want the entries before the epoch.

Reply:
Can we add a unit test to cover the bug fixed in this PR?